<h2>Why is this an issue?</h2>
<p>Merging collapsible <code>if</code> statements increases the codeâ€™s readability.</p>
<h3>Noncompliant code example</h3>
<pre>
If condition1 Then
    If condition2 Then ' Noncompliant
        ' ...
    End If
End If
</pre>
<h3>Compliant solution</h3>
<pre>
If condition1 AndAlso condition2 Then
    ' ...
End If
</pre>

