<h2>Why is this an issue?</h2>
<p>Complex boolean expressions are hard to read and so to maintain.</p>
<h3>Noncompliant code example</h3>
<p>With the default threshold value of 3</p>
<pre>
If ((condition1 AndAlso condition2) OrElse (condition3 AndAlso condition4)) AndAlso condition5) Then  'Noncompliant
  ...
End If
</pre>
<h3>Compliant solution</h3>
<pre>
If ((MyFirstCondition() OrElse MySecondCondition()) AndAlso MyLastCondition()) Then
  ...
End If
</pre>

