<p>The <code>Obsolete</code> attribute can be applied with or without arguments, but marking something <code>Obsolete</code> without including advice
as to why itâ€™s obsolete or on what to use instead will lead maintainers to waste time trying to figure those things out - every single time the
warning is encountered.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Public Class Car

    &lt;Obsolete&gt;  ' Noncompliant
    Public Sub CrankEngine(Turns As Integer)
        ' ...
    End Sub

End Class
</pre>
<h2>Compliant Solution</h2>
<pre>
Public Class Car

    &lt;Obsolete("Replaced by the automatic starter")&gt;
    Public Sub CrankEngine(Turns As Integer)
        ' ...
    End Sub

End Class
</pre>

