<h2>Why is this an issue?</h2>
<p>The <code>Obsolete</code> attribute can be applied with or without a message argument. Marking something <code>Obsolete</code> without including
advice on why itâ€™s obsolete or what to use instead will lead maintainers to waste time trying to figure those things out.</p>
<h3>Noncompliant code example</h3>
<pre>
Public Class Car

    &lt;Obsolete&gt;  ' Noncompliant
    Public Sub CrankEngine(Turns As Integer)
        ' ...
    End Sub

End Class
</pre>
<h3>Compliant solution</h3>
<pre>
Public Class Car

    &lt;Obsolete("Replaced by the automatic starter")&gt;
    Public Sub CrankEngine(Turns As Integer)
        ' ...
    End Sub

End Class
</pre>

