<h2>Why is this an issue?</h2>
<p>Declaring multiple variable on one line is difficult to read.</p>
<h3>Noncompliant code example</h3>
<pre>
Module Module1
  Public Const AAA As Integer = 5, BBB = 42, CCC As String = "foo"  ' Noncompliant
End Module
</pre>
<h3>Compliant solution</h3>
<pre>
Module Module1
  Public Const AAA As Integer = 5
  Public Const BBB = 42
  Public Const CCC as String = "foo"
End Module
</pre>

