<p>There is no good excuse for an empty class. If itâ€™s being used simply as a common extension point, it should be replaced with an
<code>Interface</code>. If it was stubbed in as a placeholder for future development it should be fleshed-out. In any other case, it should be
eliminated.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Public Class Empty ' Noncompliant

End Class
</pre>
<h2>Compliant Solution</h2>
<pre>
Public Interface IEmpty

End Interface
</pre>
<h2>Exceptions</h2>
<p>Partial classes are ignored entirely, as they are often used with Source Generators. Subclasses of System.Exception will be ignored, as even an
empty Exception class can provide useful information by its type name alone. Subclasses of certain framework types - like the PageModel class used in
ASP.NET Core Razor Pages - will also be ignored.</p>
<pre>
Imports Microsoft.AspNetCore.Mvc.RazorPages

Public Class EmptyPageModel ' Compliant - an empty PageModel can be fully functional, the VB code can be in the vbhtml file
    Inherits PageModel
End Class
</pre>

