<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all enum names match a provided regular expression.</p>
<p>The default configuration is the one recommended by Microsoft:</p>
<ul>
  <li> Pascal casing, starting with an upper case character, e.g. BackColor </li>
  <li> Short abbreviations of 2 letters can be capitalized, e.g. GetID </li>
  <li> Longer abbreviations need to be lower case, e.g. GetHtml </li>
  <li> If the enum is marked as [Flags] then its name should be plural (e.g. MyOptions), otherwise, names should be singular (e.g. MyOption) </li>
</ul>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression for non-flags enums: <code>^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?$</code></p>
<pre>
<code data-lang="vbnet">Public Enum foo ' Noncompliant
    FooValue = 0
End Enum</code>
</pre>
<p>With the default regular expression for flags enums: <code>^([A-Z]{1,3}[a-z0-9]+)*([A-Z]{2})?s$</code></p>
<pre>
<code data-lang="vbnet">&lt;Flags()&gt;
Public Enum Option ' Noncompliant
    None = 0,
    Option1 = 1,
    Option2 = 2
End Enum</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="vbnet">Public Enum Foo
    FooValue = 0
End Enum</code>
</pre>
<pre>
<code data-lang="vbnet">&lt;Flags()&gt;
Public Enum Options
    None = 0,
    Option1 = 1,
    Option2 = 2
End Enum</code>
</pre>

