<h2>Why is this an issue?</h2>
<p>The information that an enumeration type is actually an enumeration or a set of flags should not be duplicated in its name.</p>
<h3>Noncompliant code example</h3>
<pre>
Enum FooFlags ' Noncompliant
    Foo = 1
    Bar = 2
    Baz = 4
End Enum
</pre>
<h3>Compliant solution</h3>
<pre>
Enum Foo      ' Compliant
    Foo = 1
    Bar = 2
    Baz = 4
End Enum
</pre>

