<p>Array literals are more compact than array creation expressions.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="vbnet">Module Module1
    Sub Main()
        Dim foo = New String() {"a", "b", "c"} ' Noncompliant
    End Sub
End Module</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="vbnet">Module Module1
    Sub Main()
        Dim foo = {"a", "b", "c"}              ' Compliant
    End Sub
End Module</code>
</pre>

