<h2>Why is this an issue?</h2>
<p>To improve the code readability, the explicit line continuation character, <code>_</code>, should not be used. Instead, it is better to break lines
after an operator.</p>
<h3>Noncompliant code example</h3>
<pre>
Module Module1
    Sub Main()
        ' Noncompliant
        Console.WriteLine("Hello" _
                          &amp; "world")
    End Sub
End Module
</pre>
<h3>Compliant solution</h3>
<pre>
Module Module1
    Sub Main()

        Console.WriteLine("Hello" &amp;
                          "world")
    End Sub
End Module
</pre>

