<h2>Why is this an issue?</h2>
<p>Unsigned integers have different arithmetic operators than signed ones - operators that few developers understand. Therefore, signed types should
be preferred where possible.</p>
<h3>Noncompliant code example</h3>
<pre>
Module Module1
    Sub Main()
        Dim foo1 As UShort   ' Noncompliant
        Dim foo2 As UInteger ' Noncompliant
        Dim foo3 As ULong    ' Noncompliant
    End Sub
End Module
</pre>
<h3>Compliant solution</h3>
<pre>
Module Module1
    Sub Main()
        Dim foo1 As Short
        Dim foo2 As Integer
        Dim foo3 As Long
    End Sub
End Module
</pre>

