<p>The <code>... = {}</code> syntax is more compact, more readable and less error-prone.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="vbnet">Module Module1
  Sub Main()
    Dim foo(1) As String   ' Noncompliant
    foo(0) = "foo"
    foo(1) = "bar"
  End Sub
End Module</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="vbnet">Module Module1
  Sub Main()
    Dim foo = {"foo", "bar"}  ' Compliant
  End Sub
End Module</code>
</pre>

