<h2>Why is this an issue?</h2>
<p>The <code>... = {}</code> syntax is more compact, more readable and less error-prone.</p>
<h3>Noncompliant code example</h3>
<pre>
Module Module1
  Sub Main()
    Dim foo(1) As String   ' Noncompliant
    foo(0) = "foo"
    foo(1) = "bar"
  End Sub
End Module
</pre>
<h3>Compliant solution</h3>
<pre>
Module Module1
  Sub Main()
    Dim foo = {"foo", "bar"}  ' Compliant
  End Sub
End Module
</pre>

