<p>Thereâ€™s no point in creating an array solely for the purpose of passing it as a ParamArray (<code>...</code>) argument; ParamArray keyword allow to
pass a variable number of parameters that will behave exactly like an array variable inside the method implementation. Simply pass the elements
directly.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Class SurroundingClass
	Public Sub CallTheThing()
		DoTheThing(New String() {"s1", "s2"})
		DoTheThing(New String(11) {})
	End Sub

	Public Sub DoTheThing(ParamArray args As String())
		' Do something
	End Sub
End Class
</pre>
<h2>Compliant Solution</h2>
<pre>
Class SurroundingClass
	Public Sub CallTheThing()
		DoTheThing("s1", "s2")
		DoTheThing(New String(11) {})
	End Sub

	Public Sub DoTheThing(ParamArray args As String())
		' Do something
	End Sub
End Class
</pre>

