<p>Thereâ€™s no point in creating an array solely for the purpose of passing it to a <code>ParamArray</code> parameter. Simply pass the elements
directly. They will be consolidated into an array automatically.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Class SurroundingClass
    Public Sub Base()
        Method(New String() { "s1", "s2" }) ' Noncompliant: unnecessary
        Method(New String(12) {}) ' Compliant
    End Sub

    Public Sub Method(ParamArray args As String())
        ' Do something
    End Sub
End Class
</pre>
<h2>Compliant Solution</h2>
<pre>
Class SurroundingClass
    Public Sub Base()
        Method("s1", "s2")
        Method(New String(12) {})
    End Sub

    Public Sub Method(ParamArray args As String())
        ' Do something
    End Sub
End Class
</pre>

