<p>If you’re using a <code>Structure</code>, it is likely because you’re interested in performance. But by failing to implement
<code>IEquatable&lt;T&gt;</code> you’re loosing performance when comparisons are made because without <code>IEquatable&lt;T&gt;</code>, boxing and
reflection are used to make comparisons.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Structure MyStruct ' Noncompliant

    Public Property Value As Integer

End Structure
</pre>
<h2>Compliant Solution</h2>
<pre>
Structure MyStruct
    Implements IEquatable(Of MyStruct)

    Public Property Value As Integer

    Public Overloads Function Equals(other As MyStruct) As Boolean Implements IEquatable(Of MyStruct).Equals
        ' ...
    End Function

End Structure
</pre>
<h2>See</h2>
<ul>
  <li> <a href="https://learn.microsoft.com/en-us/dotnet/api/system.iequatable-1">IEquatable&lt;T&gt; Interface</a> </li>
</ul>

