<h2>Why is this an issue?</h2>
<p>Assemblies should explicitly indicate whether they are meant to be COM visible or not. If the <code>ComVisibleAttribute</code> is not present, the
default is to make the content of the assembly visible to COM clients.</p>
<p>Note that COM visibility can be overridden for individual types and members.</p>
<h3>Noncompliant code example</h3>
<pre>
Namespace MyLibrary  ' Noncompliant

End Namespace
</pre>
<h3>Compliant solution</h3>
<pre>
&lt;Assembly: Runtime.InteropServices.ComVisible(False)&gt;

Namespace MyLibrary

End Namespace
</pre>

