<h2>Why is this an issue?</h2>
<p>It is highly suspicious when a value is saved for a key or index and then unconditionally overwritten. Such replacements are likely errors.</p>
<h3>Noncompliant code example</h3>
<pre>
towns.Item(x) = "London"
towns.Item(x) = "Chicago";  // Noncompliant
</pre>

