<h2>Why is this an issue?</h2>
<p>Thereâ€™s no need to null test in conjunction with an <code>TypeOf ... Is</code> test. <code>Nothing</code> is not an instance of anything, so a null
check is redundant.</p>
<h3>Noncompliant code example</h3>
<pre>
If (x IsNot Nothing And TypeOf x Is MyClass)
    ' ...
End If

If (x Is Nothing Or TypeOf x IsNot MyClass)
    ' ...
End If
</pre>
<h3>Compliant solution</h3>
<pre>
If (TypeOf x Is MyClass)
    ' ...
End If

If (TypeOf x IsNot MyClass)
    ' ...
End If
</pre>

