<p>Creating an extension method that extends <code>Object</code> is not recommended because it makes the method available on <em>every</em> type.
Extensions should be applied at the most specialized level possible, and that is very unlikely to be <code>Object</code>.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Imports System.Runtime.CompilerServices

Module MyExtensions
    &lt;Extension&gt;
    Sub Function(obj As Object) ' Noncompliant
		' ...
    End Function
End Module
</pre>

