<h2>Why is this an issue?</h2>
<p>Creating an extension method that extends <code>Object</code> is not recommended because it makes the method available on <em>every</em> type.
Extensions should be applied at the most specialized level possible, and that is very unlikely to be <code>Object</code>.</p>
<h3>Noncompliant code example</h3>
<pre>
Imports System.Runtime.CompilerServices

Module MyExtensions
    &lt;Extension&gt;
    Sub SomeExtension(obj As Object) ' Noncompliant
        ' ...
    End Sub
End Module
</pre>

