<h2>Why is this an issue?</h2>
<p>Finalizers (historically referred to as destructors) are used to perform any necessary final clean-up when the garbage collector is collecting a
class instance. The programmer has no control over when the destructor is called; the garbage collector decides when to call it, and the destructor
implicitly calls <code>Finalize</code> on the objectâ€™s base class.</p>
<p>When you create a destructor, you should never throw an exception in it, as you are risking having your application terminated without a graceful
cleanup.</p>
<p>If a destructor throws an exception, and the runtime is not hosted by an application that overrides the default policy and the behavior of the
<code>UnhandledExceptionEventHandler</code>, then the runtime terminates the process immediately without graceful cleanup (<code>finally</code> blocks
and <code>Finalizer</code> methods are not executed).</p>
<p>The rule raises an issue on <code>throw</code> statements used in destructors.</p>
<h3>Noncompliant code example</h3>
<pre data-diff-id="1" data-diff-type="noncompliant">
class MyClass
{
    ~MyClass()
    {
        throw new NotImplementedException(); // Noncompliant
    }
}
</pre>
<h3>Compliant solution</h3>
<pre data-diff-id="1" data-diff-type="compliant">
class MyClass
{
    ~MyClass()
    {

    }
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> <a href="https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/finalizers">Finalizers (destructors)</a> </li>
  <li> <a href="https://learn.microsoft.com/en-us/dotnet/api/system.appdomain">App Domain</a> </li>
  <li> <a href="https://learn.microsoft.com/en-us/dotnet/api/system.appdomain.unhandledexception">AppDomain.UnhandledException Event</a> </li>
</ul>

