<h2>Why is this an issue?</h2>
<p>While not technically incorrect, the omission of curly braces can be misleading, and may lead to the introduction of errors during maintenance.</p>
<h3>Noncompliant code example</h3>
<pre>
// the two statements seems to be attached to the if statement, but that is only true for the first one:
if (condition)
  ExecuteSomething();
  CheckSomething();
</pre>
<h3>Compliant solution</h3>
<pre>
if (condition)
{
  ExecuteSomething();
  CheckSomething();
}
</pre>

