<p>The use of non-short-circuit logic in a boolean context is likely a mistake - one that could cause serious program errors as conditions are
evaluated under the wrong circumstances.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="csharp">if (GetTrue() | GetFalse()) // Noncompliant; both sides evaluated
{
}</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="csharp">if (GetTrue() || GetFalse()) // true short-circuit logic
{
}</code>
</pre>

