<h2>Why is this an issue?</h2>
<p>The use of non-short-circuit logic in a boolean context is likely a mistake - one that could cause serious program errors as conditions are
evaluated under the wrong circumstances.</p>
<h3>Noncompliant code example</h3>
<pre>
if (GetTrue() | GetFalse()) // Noncompliant; both sides evaluated
{
}
</pre>
<h3>Compliant solution</h3>
<pre>
if (GetTrue() || GetFalse()) // true short-circuit logic
{
}
</pre>

