<p>Certain mathematical comparisons will always return the same value, and should simply not be made.</p>
<p>These comparisons will return either always <code>true</code> or always <code>false</code> depending on the kind of comparison:</p>
<ul>
  <li> Comparing a <code>char</code> with a numeric constant that is outside of the range of <code>char</code>. </li>
  <li> Comparing a <code>float</code> with a numeric constant that is outside of the range of <code>float</code>. </li>
  <li> Comparing a <code>long</code> with a numeric constant that is outside of the range of <code>long</code>. </li>
  <li> Comparing a <code>ulong</code> with a numeric constant that is outside of the range of <code>ulong</code>. </li>
</ul>
<h2>Noncompliant Code Example</h2>
<pre>
float f = 42.0f;
if (f &lt;= double.MaxValue) { } // Noncompliant
</pre>

