<p>This rule is deprecated; use {rule:csharpsquid:S106} instead.</p>
<h2>Why is this an issue?</h2>
<p>Debug statements are always useful during development. But include them in production code - particularly in code that runs client-side - and you
run the risk of inadvertently exposing sensitive information.</p>
<h3>Noncompliant code example</h3>
<pre>
private void DoSomething()
{
    // ...
    Console.WriteLine("so far, so good..."); // Noncompliant
    // ...
}
</pre>
<h3>Exceptions</h3>
<p>The following are ignored by this rule:</p>
<ul>
  <li> Console Applications </li>
  <li> Calls in methods decorated with <code>[Conditional ("DEBUG")]</code> </li>
  <li> Calls included in DEBUG preprocessor branches (<code>#if DEBUG</code>) </li>
</ul>
<h2>Resources</h2>
<ul>
  <li> <a href="https://www.owasp.org/www-project-top-ten/2017/A3_2017-Sensitive_Data_Exposure">OWASP Top 10 2017 Category A3</a> - Sensitive Data
  Exposure </li>
</ul>

