<h2>Why is this an issue?</h2>
<p>Since C# 5.0, <code>async</code> and <code>await</code> are contextual keywords. Contextual keywords do have a particular meaning in some contexts,
but can still be used as variable names. Keywords, on the other hand, are always reserved, and therefore are not valid variable names. To avoid any
confusion though, it is best to not use <code>async</code> and <code>await</code> as identifiers.</p>
<h3>Noncompliant code example</h3>
<pre>
int await = 42; // Noncompliant
</pre>
<h3>Compliant solution</h3>
<pre>
int someOtherName = 42;
</pre>

