<h2>Why is this an issue?</h2>
<p>Namespaces with no lines of code clutter a project and should be removed.</p>
<h3>Noncompliant code example</h3>
<pre>
namespace MyEmptyNamespace // Noncompliant
{

}
</pre>

