<p>Namespaces with no lines of code clutter a project and should be removed.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="text">namespace MyEmptyNamespace // Noncompliant
{

}</code>
</pre>

