<h2>Why is this an issue?</h2>
<p>A class with only <code>private</code> constructors canâ€™t be instantiated, thus, it seems to be pointless code.</p>
<h3>Noncompliant code example</h3>
<pre>
public class MyClass // Noncompliant
{
  private MyClass() { ... }
}
</pre>
<h3>Compliant solution</h3>
<pre>
public class MyClass
{
  public MyClass() { ... }
}
</pre>
<h3>Exceptions</h3>
<p>Classes that themselves access their private constructors (singletons or smart enums) are ignored. Classes with only <code>static</code> members
are also ignored because they are covered by Rule {rule:csharpsquid:S1118}.</p>

