<p>The <code>default</code> clause should take appropriate action. Having an empty <code>default</code> is a waste of keystrokes.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="csharp">enum Fruit
{
  Apple,
  Orange,
  Banana
}

void PrintName(Fruit fruit)
{
  switch(fruit)
  {
    case Fruit.Apple:
      Console.WriteLine("apple");
      break;
    default:  //Noncompliant
      break;
  }
}</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="csharp">enum Fruit
{
  Apple,
  Orange,
  Banana
}

void PrintName(Fruit fruit)
{
  switch(fruit)
  {
    case Fruit.Apple:
      Console.WriteLine("apple");
      break;
    default:
      throw new NotSupportedException();
  }
}</code>
</pre>
<p>or</p>
<pre>
<code data-lang="csharp">void PrintName(Fruit fruit)
{
  switch(fruit)
  {
    case Fruit.Apple:
      Console.WriteLine("apple");
      break;
  }
}</code>
</pre>
<h2>Exceptions</h2>
<p><code>default</code> clauses containing only a comment are ignored with the assumption that they are empty on purpose and the comment documents
why.</p>

