<h2>Why is this an issue?</h2>
<p>When declaring a Windows Communication Foundation (WCF) <code>OperationContract</code> method one-way, that service method won’t return any result,
not even an underlying empty confirmation message. These are fire-and-forget methods that are useful in event-like communication. Specifying a return
type therefore does not make sense.</p>
<h3>Noncompliant code example</h3>
<pre>
[ServiceContract]
interface IMyService
{
  [OperationContract(IsOneWay = true)]
  int SomethingHappened(int parameter); // Noncompliant
}
</pre>
<h3>Compliant solution</h3>
<pre>
[ServiceContract]
interface IMyService
{
  [OperationContract(IsOneWay = true)]
  void SomethingHappened(int parameter);
}
</pre>
<h3>Exceptions</h3>
<p>The rule doesn’t report if <code>OperationContractAttribute.AsyncPattern</code> is set to <code>true</code>.</p>

