<h2>Why is this an issue?</h2>
<p>Strings and integral types are typically used as indexers. When some other type is required, it typically indicates design problems, and
potentially a situation where a method should be used instead.</p>
<h3>Noncompliant code example</h3>
<pre>
public int this[MyCustomClass index]  // Noncompliant
{
    // get and set accessors
}
</pre>

