<p>Finalizers come with a performance cost due to the overhead of tracking the life cycle of objects. An empty one is consequently costly with no
benefit or justification.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="csharp">public class Foo
{
    ~Foo() // Noncompliant
    {
    }
}</code>
</pre>

