<h2>Why is this an issue?</h2>
<p>Finalizers come with a performance cost due to the overhead of tracking the life cycle of objects. An empty one is consequently costly with no
benefit or justification.</p>
<h3>Noncompliant code example</h3>
<pre>
public class Foo
{
    ~Foo() // Noncompliant
    {
    }
}
</pre>

