<h2>Why is this an issue?</h2>
<p>Assemblies should explicitly indicate whether they are meant to be COM visible or not. If the <code>ComVisibleAttribute</code> is not present, the
default is to make the content of the assembly visible to COM clients.</p>
<p>Note that COM visibility can be overridden for individual types and members.</p>
<h3>Noncompliant code example</h3>
<pre>
using System;

namespace MyLibrary  // Noncompliant
{
}
</pre>
<h3>Compliant solution</h3>
<pre>
using System;

[assembly: System.Runtime.InteropServices.ComVisible(false)]
namespace MyLibrary
{
}
</pre>

