<h2>Why is this an issue?</h2>
<p>When a method in a derived class has the same name as a method in the base class but with a signature that only differs by types that are weakly
derived (e.g. <code>object</code> vs <code>string</code>), the result is that the base method becomes hidden.</p>
<h3>Noncompliant code example</h3>
<pre>
using System;

namespace MyLibrary
{
  class Foo
  {
    internal void SomeMethod(string s1, string s2) { }
  }

  class Bar : Foo
  {
    internal void SomeMethod(string s1, object o2) { }  // Noncompliant
  }
}
</pre>
<h3>Compliant solution</h3>
<pre>
using System;

namespace MyLibrary
{
  class Foo
  {
    internal void SomeMethod(string s1, string s2) { }
  }

  class Bar : Foo
  {
    internal void SomeOtherMethod(string s1, object o2) { }
  }
}
</pre>

