<h2>Why is this an issue?</h2>
<p>A nested type is a type argument that is also a generic type. Calling a method with such a nested type argument requires complicated and confusing
code. It should be avoided as much as possible.</p>
<h3>Noncompliant code example</h3>
<pre>
using System;
using System.Collections.Generic;

namespace MyLibrary
{
  public class Foo
  {
    public void DoSomething(ICollection&lt;ICollection&lt;int&gt;&gt; outerCollect) // Noncompliant
    {
    }
  }
}
</pre>

