<h2>Why is this an issue?</h2>
<p>Empty interfaces are usually used as a marker or a way to identify groups of types. The preferred way to achieve this is to use custom
attributes.</p>
<h3>Noncompliant code example</h3>
<pre>
using System;

namespace MyLibrary
{
   public interface MyInterface // Noncompliant
   {
   }
}
</pre>
<h3>Compliant solution</h3>
<pre>
using System;

namespace MyLibrary
{
   public interface MyInterface
   {
      void Foo();
   }
}
</pre>

