<p>Empty interfaces are usually used as a marker or a way to identify groups of types. The preferred way to achieve this is to use custom
attributes.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="text">using System;

namespace MyLibrary
{
   public interface MyInterface // Noncompliant
   {
   }
}</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="text">using System;

namespace MyLibrary
{
   public interface MyInterface
   {
      void Foo();
   }
}</code>
</pre>

