<h2>Why is this an issue?</h2>
<p>ï»¿Creating an extension method that extends <code>object</code> is not recommended because it makes the method available on <em>every</em> type.
Extensions should be applied at the most specialized level possible, and that is very unlikely to be <code>object</code>.</p>
<h3>Noncompliant code example</h3>
<pre>
public static class MyExtensions
{
    public static void SomeExtension(this object obj) // Noncompliant
    {
        // ...
    }
}
</pre>

