<h2>Why is this an issue?</h2>
<p>Returning <code>null</code> from a non-<code>async</code> <code>Task</code>/<code>Task&lt;T&gt;</code> method will cause a
<code>NullReferenceException</code> at runtime. This problem can be avoided by returning <code>Task.FromResult&lt;T&gt;(null)</code> instead.</p>
<h3>Noncompliant code example</h3>
<pre>
public Task&lt;object&gt; GetFooAsync()
{
    return null; // Noncompliant
}
</pre>
<h3>Compliant solution</h3>
<pre>
public Task&lt;object&gt; GetFooAsync()
{
    return Task.FromResult&lt;object&gt;(null);
}
</pre>

