<h2>Why is this an issue?</h2>
<p>Using upper case literal suffixes removes the potential ambiguity between "1" (digit 1) and "l" (letter el) for declaring literals.</p>
<h3>Noncompliant code example</h3>
<pre>
const long b = 0l;      // Noncompliant
</pre>
<h3>Compliant solution</h3>
<pre>
const long b = 0L;
</pre>

