<p>When <code>switch</code> statements have large sets of case clauses, it is usually an attempt to map two sets of data. A <code>Dictionary</code>
should be used instead to make the code more readable and maintainable.</p>
<h2>Noncompliant Code Example</h2>
<p>With a “Maximum number of case” set to 14</p>
<pre>
public class TooManyCase
{
    public int switchCase(char ch)
    {
        switch(ch) {  // Noncompliant
            case 'a':
            case 'b':
            case 'c':
            case 'd':
            case 'e':
                return 1;
            case 'f':
            case 'g':
            case 'h':
            case 'i':
            case 'j':
                return 2;
            case 'k':
            case 'l':
            case 'm':
            case 'n':
            case 'o':
                return 3;
            default:
                return 4;
        }
    }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
using System.Collections.Generic;

public class TooManyCase
{
    Dictionary&lt;char, int&gt; matching = new Dictionary&lt;char, int&gt;()
    {
        {'a', 1}, {'b', 1}, {'c', 1}, {'d', 1}, {'e', 1},
        {'f', 2}, {'g', 2}, {'h', 2}, {'i', 2}, {'j', 2},
        {'k', 3}, {'l', 3}, {'m', 3}, {'n', 3}, {'o', 3}
    };

    public int withDictionary(char ch)
    {
        int value;
        if (this.matching.TryGetValue(ch, out value)) {
            return value;
        } else {
            return 4;
        }
    }
}
</pre>
<h2>Exceptions</h2>
<p>This rule ignores <code>switch</code>es over <code>Enum</code>s and empty, fall-through cases.</p>

