<p>Having an unconditional <code>break</code>, <code>return</code>, <code>(@)throw</code> or <code>goto</code> in a loop renders it useless; the loop
will only execute once and the loop structure itself is simply wasted keystrokes.</p>
<p>Having an unconditional <code>continue</code> in a loop can render the loop meaningless, or is itself wasted keystrokes, depending on where in the
loop it occurs.</p>
<p>Having an unconditional <code>return</code> anywhere other than at the end of a function or method simply renders all the rest of the code in the
method useless.</p>
<p>For these reasons, unconditional jump statements should never be used except for the final <code>return</code> in a function or method.</p>
<h2>Noncompliant Code Example</h2>
<pre>
for (var i = 0; i &lt; 10; i++)
{
    Console.WriteLine(i);
    break;  // loop only executes once
}

for (var i = 0; i &lt; 10; i++)
{
    continue;
    Console.WriteLine(i); // this is never executed
}

for (var i = 0; i &lt; 10; i++)
{
    Console.WriteLine(i);
    continue;  // this is meaningless; the loop would continue anyway
}
</pre>
<h2>Compliant Solution</h2>
<pre>
for (var i = 0; i &lt; 10; i++)
{
    Console.WriteLine(i);
}

for (var i = 0; i &lt; 10; i++)
{
    Console.WriteLine(i);
    if (ErrorOccurred())
    {
        break;
    }
}
</pre>
<h2>See</h2>
<ul>
  <li> MISRA C:2004, 14.1 - There shall be no unreachable code. </li>
  <li> MISRA C++:2008, 0-1-1 - A <em>project</em> shall not contain <em>unreachable code</em>. </li>
  <li> MISRA C++:2008, 0-1-9 - There shall be no dead code. </li>
  <li> MISRA C:2012, 2.2 - There shall be no dead code </li>
  <li> <a href="https://www.securecoding.cert.org/confluence/x/NYA5">CERT, MSC12-C.</a> - Detect and remove code that has no effect or is never
  executed </li>
  <li> <a href="https://www.securecoding.cert.org/confluence/x/SIIyAQ">CERT, MSC12-CPP.</a> - Detect and remove code that has no effect </li>
</ul>

