<p>Delegate event handlers should have a very specific signature:</p>
<ul>
  <li> Return type <code>void</code>. </li>
  <li> First argument of type <code>System.Object</code> and named 'sender'. </li>
  <li> Second argument of type <code>System.EventArgs</code> and is named 'e'. </li>
</ul>
<p>This rule raises an issue whenever a <code>delegate</code> declaration doesn't match that signature.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public delegate void AlarmEventHandler(object s);  // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
public delegate void AlarmEventHandler(object sender, AlarmEventArgs e);
</pre>
<h2>See</h2>
<p><a href="https://msdn.microsoft.com/en-us/library/edzehd2t.aspx">Handling and Raising Events</a></p>

