<p>The size of a collection and the length of an array are always greater than or equal to zero. So testing that a size or length is greater than or
equal to zero doesn't make sense, since the result is always <code>true</code>. Perhaps the intent was to check the non-emptiness of the collection or
array instead. </p>
<h2>Noncompliant Code Example</h2>
<pre>
if(something.size() &gt;= 0){...}

boolean result = something.length() &gt;=0;

if(something.length &gt;= 0){...}
</pre>

