<p>Merging collapsible <code>if</code> statements increases the codeâ€™s readability.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="vbnet">If condition1 Then
    If condition2 Then ' Noncompliant
        ' ...
    End If
End If</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="vbnet">If condition1 AndAlso condition2 Then
    ' ...
End If</code>
</pre>

