<p><code>switch</code> statements are useful when there are many different cases depending on the value of the same expression.</p>
<p>For just one or two cases however, the code will be more readable with <code>if</code> statements.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="vbnet">Select Case variable
    Case 0
        doSomething()
    Case Else
        doSomethingElse()
End Select</code>
</pre>
<h2>Compliant Solution</h2>
<pre>
<code data-lang="vbnet">If variable = 0 Then
    doSomething()
Else
    doSomethingElse()
End If</code>
</pre>

