<p>This rule allows you to track the usage of the <code>SuppressMessage</code> attributes and <code>#pragma warning disable</code> mechanism.</p>
<h2>Noncompliant Code Example</h2>
<pre>
<code data-lang="csharp">[SuppressMessage("", "S100")]
...

#pragma warning disable S100
...
#pragma warning restore S100</code>
</pre>

