<p>Having two <code>Cases</code> in the same <code>Select</code> statement or branches in the same <code>If</code> structure with the same
implementation is at best duplicate code, and at worst a coding error. If the same logic is truly needed for both instances, then in an
<code>If</code> structure they should be combined, or for a <code>Select</code>, one should fall through to the other.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Select i
  Case 1
    DoSomething()
  Case 2
    DoSomethingDifferent()
  Case 3  ' Noncompliant; duplicates case 1's implementation
    DoSomething()
  Case Else:
    DoTheRest()
End Select

If a &gt;= 0 AndAlso a &lt; 10 Then
  DoTheThing()
ElseIf a &gt;= 10 AndAlso a &lt; 20 Then
  DoTheOtherThing()
ElseIf a &gt;= 20 AndAlso a &lt; 50
  DoTheThing() ' Noncompliant; duplicates first condition
Else
  DoTheRest();
End If
</pre>
<h2>Compliant Solution</h2>
<pre>
Select i
  Case 1, 3
    DoSomething()
  Case 2
    DoSomethingDifferent()
  Case Else
    DoTheRest()
End Select

If (a &gt;= 0 AndAlso a &lt; 10) OrElse (a &gt;= 20 AndAlso a &lt; 50) Then
  DoTheThing()
ElseIf a &gt;= 10 AndAlso a &lt; 20 Then
  DoTheOtherThing()
Else
  DoTheRest();
End If
</pre>
<p>or</p>
<pre>
Select i
  Case 1
    DoSomething()
  Case 2
    DoSomethingDifferent()
  Case 3
    DoTheThirdThing()
  Case Else
    DoTheRest()
End Select

If a &gt;= 0 AndAlso a &lt; 10 Then
  DoTheThing()
ElseIf a &gt;= 10 AndAlso a &lt; 20 Then
  DoTheOtherThing()
ElseIf a &gt;= 20 AndAlso a &lt; 50
  DoTheThirdThing()
Else
  DoTheRest();
End If
</pre>
<h2>Exceptions</h2>
<p>Blocks in an <code>if</code> chain that contain a single line of code are ignored, as are blocks in a <code>switch</code> statement that contain a
single line of code with or without a following <code>break</code>.</p>

