<p>For clarity, all overloads of the same method should be grouped together. That lets both users and maintainers quickly understand all the current
available options.</p>
<h2>Noncompliant Code Example</h2>
<pre>
interface IMyInterface
{
  int DoTheThing();
  string DoTheOtherThing(); // Noncompliant
  int DoTheThing(string s);
}
</pre>
<h2>Compliant Solution</h2>
<pre>
interface IMyInterface
{
  int DoTheThing();
  int DoTheThing(string s);
  string DoTheOtherThing();
}
</pre>

