<p>When two methods have the same implementation, either it was a mistake - something else was intended - or the duplication was intentional, but may
be confusing to maintainers. In the latter case, one implementation should invoke the other.</p>
<h2>Noncompliant Code Example</h2>
<pre>
private final static String CODE = "bounteous";

public String getCode() {
  return CODE;
}

public String getName() {  // Noncompliant
  return CODE;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
private final static String CODE = "bounteous";

public String getCode() {
  return CODE;
}

public String getName() {
  return getCode();
}
</pre>
<h2>Exceptions</h2>
<p>Empty methods that only <code>return void;</code> and methods that simply throw an <code>UnsupportedOperationException()</code> or
<code>NotImplementedException</code> are ignored.</p>

