<p>Because parameter names could be changed during refactoring, they should not be spelled out literally in strings. Instead, use
<code>NameOf()</code>, and the string that's output will always be correct.</p>
<p>This rule raises an issue when any string in the <code>Throw</code> statement is an exact match for the name of one of the method parameters.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Public Sub DoSomething(param As Integer)
    If (param &lt; 0)
        Throw New Exception("param") ' Noncompliant
    End If
End Sub
</pre>
<h2>Compliant Solution</h2>
<pre>
Public Sub DoSomething(param As Integer)
    If (param &lt; 0)
        Throw New Exception(NameOf(param)) ' Noncompliant
    End If
End Sub
</pre>
<h2>Exceptions</h2>
<p>The rule doesn't raise any issue when using VB.NET &lt; 14.0.</p>

